rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -1447,10 +1594,23 @@ def batch_predict(
             out["i_scores"] = scores
 
         if self.config[ENTITY_RECOGNITION]:
-            logits = self._tf_layers["embed.logits"](text_transformed)
-            pred_ids = self._tf_layers["crf"](logits, sequence_lengths - 1)
-            out["e_ids"] = pred_ids
-
+            prev_logits = None
+            for name in CRF_ORDER:
+                if name not in self._crf_num_tags or self._crf_num_tags[name] == 0:
+                    continue
+
+                _input = text_transformed
+                if prev_logits is not None:
+                    _input = tf.concat([_input, prev_logits], axis=-1)
+
+                _logits = self._tf_layers[f"embed.{name}.logits"](_input)
+                pred_ids = self._tf_layers[f"crf.{name}"](_logits, sequence_lengths - 1)
```

Comment:
I think this part can be united with similar part in `_calculate_entity_loss` into a separate helper